src/shared/analytics/google-analytics.ts

```diff
@@ -0,0 +1,106 @@
+import browser from 'webextension-polyfill';
+import * as browserStorage from 'src/background/webapis/storage';
+import {
+  GOOGLE_ANALYTICS_API_SECRET,
+  GOOGLE_ANALYTICS_MEASUREMENT_ID,
+} from 'src/env/config';
+import { Loglevel, logTable, logToConsole } from '../logger';
+import { onIdle } from './analytics';
+
+type GoogleAnalyticsEvent = 'page_view';
+
+export const SESSION_ID_KEY = 'GOOGLE_ANALYTICS_V4_SESSION_ID';
+const CLIENT_ID_KEY = 'GOOGLE_ANALYTICS_V4_CLIENT_ID';
+const ENDPOINT = 'https://www.google-analytics.com/mp/collect';
+const DEFAULT_ENGAGEMENT_TIME_IN_MSEC = 100;
+
+function readClientid() {
+  return browserStorage.get<string>(CLIENT_ID_KEY);
+}
+
+async function getClientId() {
+  let clientId = await readClientid();
+  if (!clientId) {
+    // Generate a unique client ID for Google Analytics V4.
+    // The actual value is not relevant.
+    clientId = crypto.randomUUID();
+    await browserStorage.set(CLIENT_ID_KEY, clientId);
+  }
+  return clientId;
+}
+
+export async function resetGoogleAnalyticsSessionId() {
+  const sessionId = crypto.randomUUID() as string;
+  await chrome.storage.session.set({
```

Comment:
Isn't this also `await browserStorage.set(CLIENT_ID_KEY, clientId);`?

Reply:
`local` and `session` storage are different (according to [docs](https://developer.chrome.com/docs/extensions/reference/api/storage#storage-areas)). Probably we could use browserStorage (`local`) here as well, but the google analytics v4 docs suggests using session storage. I think that is because the session storage is cleared if the extension is disabled, reloaded or updated and when the browser restarts